### PR TITLE
Update to latest version of Maven 3.x

### PR DESCRIPTION
**Pull request by bigdaz**
Maven 3.6.1 brings in a dependency with a critical security vulnerability.
This version update should bring no behaviour change.

I'm not sure what I need to do to ensure the updates to verification-metadata are OK.

### CODE REVIEW

**Review by ov7a — APPROVED** [review]

**Comment by bigdaz** [comment]
@bot-gradle test and merge

File: subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/MavenProjectsCreator.java [hunk 1]
```diff
@@ -50,6 +50,7 @@
 import java.util.Properties;
 import java.util.Set;
 
+@SuppressWarnings("deprecation") // This Class uses a number of types deprecated in Maven
```
[the remark this answers is not in the document]
Comment (ljacomet) [reply subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/MavenProjectsCreator.java:53]: I'll file an issue, but none of this code is going away before Maven 4.

By looking at Maven code, there are no indications on what should be used instead.

**Comment by ljacomet** [comment]
@bot-gradle test this

File: subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/MavenProjectsCreator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ljacomet) [reply subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/MavenProjectsCreator.java:53]: Filed #25932

[the remark this answers is not in the document]
Comment (bigdaz) [reply subprojects/build-init/src/main/java/org/gradleinternal/buildinit/plugins/internal/maven/MavenProjectsCreator.java:53]: Note that there were already 2 deprecation warnings emitted with Maven 3.6.3: the upgrade added 1 new warning.
The main reason I moved this up to a Class level suppression was that the existing code uses `forEach` in a way that didn't permit a localized suppression.

**Comment by ljacomet** [comment]
Test failures come from a POM comparison and we now see `artifactId` before `groupId` in dependencies.
This can probably be solved by editing some expected metadata files in `subprojects/smoke-test/src/smokeTest/resources/org/gradle/smoketests/gmm-example/expected-metadata` or making the comparison ignore those.

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by bigdaz** [comment]
@bot-gradle test this

**Comment by bigdaz** [comment]
@bot-gradle test and merge

**Comment by bigdaz** [comment]
@ljacomet I pushed a change to the comparison POM files. The `groupId` elements now precede `artifactId` within an `<exclusion>`.

**Comment by bigdaz** [comment]
Woot!!!
@ljacomet another critical security alert fixed: https://github.com/gradle/gradle/security/dependabot/254

**Pull request merged**